webpack.config.js

```diff
@@ -23,14 +23,11 @@ module.exports = {
         use: [{ loader: 'babel-loader' }],
       },
       {
-        test: /\.s[ac]ss$/i,
+        test: /\.css$/,
         use: [
-          // Creates `style` nodes from JS strings
           'style-loader',
-          // Translates CSS into CommonJS
-          'css-loader',
-          // Compiles Sass to CSS
-          'sass-loader',
+          { loader: 'css-loader', options: { importLoaders: 1 } },
```

Comment:
Wouldn't it be better to extract CSS here instead of using `npm run build:css`? 

```
        test: /\.css$/,
        use: [
          DEV ? 'style-loader' : MiniCssExtractPlugin.loader,
          { loader: 'css-loader', options: { importLoaders: 1 } },
          'postcss-loader',
        ],
```

Reply:
Not really. It's a library. Even after build a developer should be able to read source. Minifier always set up on the end projects.

Reply:
AFAIK it's not minifier. it should do just the work of `npm run build:css` for you automatically. 

> This plugin extracts CSS into separate files. It creates a CSS file per JS file which contains CSS. It supports On-Demand-Loading of CSS and SourceMaps.

Reply:
I do not completely understand this. As far as I know we use Webpack only because of react-stylguide and for development server. However, the production build goes with simple NPM scripts.

To extend, I am against building with Webpack so I do not mind this only for docs.

I payed attention to `MiniCssExtractPlugin.loader` only, sorry. I think we do not need this. However, if talk about style loader, it's good for some cases but not here because we really need it separate.

My idea was to allow user do the following:
```js
import { Icon, Button } from '@oacore/design'
import '@oacore/design/index.css'
```
along with
```js
import Icon from '@oacore/design/icon'
import '@oacore/design/icon/index.css'
```

In the first case a user imports whole library (because he knows that he is using whole library) with all the css in a single file. In the second case the user imports only _Icon_ with its CSS only because this is all he/she needs. To give an ability to choose I generate many CSS/JS files and I think this is good approach.

Although, I personally do not really like how it look now so we have to explore it more. Do you have any opinion?

Reply:
Well, I thought we would use webpack for the building process too. We need to transpile the code and build one module from it. Currently, it is done with `npm run start` script. 

We have the same idea the only thing I am not sure about is whether this is possible 
`import Icon from '@oacore/design/icon'` I am not sure how to do it. The only solution I am aware of is creating sub-packages




.

Reply:
Not really. It's like `import Link from 'next/link'`. Link is not a sub-package. It's possible to resolve package subdirectories but we have to find a way to publish it right.

Reply:
About Webpack, we do not have to. We do not need to use Webpack twice. Webpack on the final project will resolve all dependencies and bundle everything once. We just need to transpile JS and CSS because Babel and other tools usually skip `node_modules`.

Reply:
But all the libraries used here will need to be peer dependencies in other packages, won't be? The ideal scenario would be to have only two peer dependencies react and react-dom. Everything else should be installed here. Maybe I am getting it wrong. If you know what you are doing then go for it and try to make a minimal example for it.

Reply:
No, `dependencies` are always installed if it's listed but `devDependencies` skipped. `peerDependencies` throw warning if absent.

Reply:
I could not find a way to avoid `dist` in module path. There are a few options, I created #14 to discuss this.